src/GameLogic/PlayerActions/Craftings/RefineStoneCrafting.cs

```diff
@@ -114,7 +128,10 @@ public RefineStoneCrafting(SimpleCraftingSettings settings)
     protected override bool RequiredItemMatches(Item item, ItemCraftingRequiredItem requiredItem)
     {
         return base.RequiredItemMatches(item, requiredItem)
-               && (item.IsWearable() && !this._excludedItems.Contains((item.Definition!.Group, item.Definition.Number)));
+               && item.IsWearable()
+               && !item.IsAncient()
+               && !item.ItemOptions.Any(o => o.ItemOption?.OptionType == ItemOptionTypes.HarmonyOption)
+               && (!this._excludedItems.Contains((item.Definition!.Group, item.Definition.Number)) || item.Level > 3);
```

Comment:
Is this line correct?
Shouldn't it be `&& !this._excludedItems.Contains((item.Definition!.Group, item.Definition.Number) && item.Level > 3;` ?

Reply:
I believe the point of the exclusion list was to stop you from using cheap, merchant-store-available items. Therefore, you either want the used item not to be in the list, or, if it is, you want it to be level 4 or higher (because they are sold at level 3). Check the quoted reference below and [here](https://github.com/kyleruss/emu-server/blob/202856a74c905c203b9b2795fd161f564ca8b257/GameServer/Source/JewelOfHarmonySystem.cpp#L918).

> Updated excluded items list for Lower Refining Stone crafting ([reference](https://github.com/kyleruss/emu-server/blob/202856a74c905c203b9b2795fd161f564ca8b257/Template/zGameServer/Data/Item/JewelOfHarmonySmelt.txt))